src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -361,6 +361,72 @@ def _find_or_create_unresolved_symbol(location, name, scope_limit=None,
+    the necessary code to the parent module.
+
+    :returns: the comparison interface symbol.
+    :rtype: :py:class:`psyclone.psyir.symbols.Symbol`
+
+    :raises NotImplementedError: if there is no ancestor module container
+        on which to add the interface code into.
+    '''
+    try:
+        return node.scope.symbol_table.lookup_with_tag("psyclone_internal_cmp")
+    except KeyError:
+        container = node.ancestor(Container)
+        if container and not isinstance(container, FileContainer):
+            from psyclone.psyir.frontend.fortran import FortranReader
+            name_interface = node.scope.symbol_table.next_available_name(
+                                                       "psyclone_internal_cmp")
+            name_f_int = node.scope.symbol_table.next_available_name(
+                                                       "psyclone_cmp_int")
+            name_f_logical = node.scope.symbol_table.next_available_name(
+                                                       "psyclone_cmp_logical")
+            name_f_char = node.scope.symbol_table.next_available_name(
+                                                       "psyclone_cmp_char")
+            fortran_reader = FortranReader()
+            dummymod = fortran_reader.psyir_from_source(f'''
+            module dummy
+                implicit none
+                interface {name_interface}
+                    procedure {name_f_int}
+                    procedure {name_f_logical}
+                    procedure {name_f_char}
+                end interface {name_interface}
+                private {name_interface}
+                private {name_f_int}, {name_f_logical}, {name_f_char}
+                contains
+                logical pure function {name_f_int}(op1, op2)
+                    integer, intent(in) :: op1, op2
+                    {name_f_int} = op1.eq.op2
+                end function
+                logical pure function {name_f_logical}(op1, op2)
+                    logical, intent(in) :: op1, op2
+                    {name_f_logical} = op1.eqv.op2
+                end function
+                logical pure function {name_f_char}(op1, op2)
+                    character(*), intent(in) :: op1, op2
+                    {name_f_char} = op1.eq.op2
+                end function
+            end module dummy
+            ''').children[0]  # We skip the top FileContainer
+
+            # Add the new functions and interface to the ancestor container
+            container.children.extend(dummymod.pop_all_children())
+            container.symbol_table.merge(dummymod.symbol_table)
+            symbol = container.symbol_table.lookup(name_interface)
+            # Add the appropriate tag to find it regardless of the name
+            container.symbol_table.tags_dict['psyclone_internal_cmp'] = symbol
+            return symbol
+
+    raise NotImplementedError(
+        "Could not find the generic comparison interface nor an ancestor "
```

Comment:
Perhaps: "...interface and {node.debug_string()} does not have an ancestor Container in which to add it."

Reply:
I have improved the message, but not added node.debug_string() because the node is not fully formed yet and it misses mandatory children that make the debug_string fail. I think this needs the "origin_string" which points to the original source file line, but I am doing this in another branch and is not ready yet.

(Note that this ends up in a Codeblock and not an error message)